modules/volunteers/admin/sitesupport/giftTable.js

```diff
@@ -260,6 +244,57 @@ export default {
         return [u.length, nonGroup];
       }
       return [0, 1];
+    },
+    sortTable(column) {
+      if (column.unsortable) {
+        return;
+      }
+      if (column.value === this.sortColumn) {
+        this.decendingSort = !this.decendingSort;
+      }
+      this.sortColumn = column.value;
+      let targetLeft = (this.decendingSort) ? 1 : -1;
+      let targetRight  = (this.decendingSort) ? -1 : 1;
+
+      this.records.sort((a,b) => {
+        let aValue = a[column.value];
+        let bValue = b[column.value];
+        if (column.value == 'remaining') {
+          aValue = a.inventory - a.claimed;
+          bValue = b.inventory - b.claimed;
+        } else if (column.type === Number) {
+          aValue = parseFloat(aValue);
+          bValue = parseFloat(bValue);
+        }
+
+        if (aValue === bValue) {
+          if (a.name ===  b.name) {
+            return ((parseFloat(a.value) > parseFloat(b.value)) ? targetLeft : targetRight);
+          }
+          return ((a.name > b.name) ? targetLeft : targetRight);
+        }
+        return ((aValue > bValue) ? targetLeft : targetRight);
+      });
+
+      this.records.sort((a,b) => {
+        let aValue = a[column.value];
+        let bValue = b[column.value];
+        if (column.value == 'remaining') {
+          aValue = a.inventory - a.claimed;
+          bValue = b.inventory - b.claimed;
+        } else if (column.type === Number) {
+          aValue = parseFloat(aValue);
+          bValue = parseFloat(bValue);
+        }
+
+        if (a.reward_group !== null && b.reward_group === null) {return -1;}
+        if (b.reward_group !== null && a.reward_group === null) {return 1;}
+        if (b.reward_group === null && a.reward_group === null) {return 0;}
+        if (parseInt(b.reward_group.id) === parseInt(a.reward_group.id)) {
+          return ((aValue > bValue) ? targetLeft : targetRight);
```

Comment:
If this is the only piece that should be sortable, consider the comment above about `sortDirection` and we could simplify this to `return (aValue - bValue) * sortDirection;`